scripts/peer/peer.py

```diff
@@ -0,0 +1,139 @@
+"""
+Test run this with:
+
+    cd <dir where peer.py file is>
+    analysis-runner \
+        --access-level test \
+        --description 'Test run peer analysis' \
+        --output-dir 'kat/2022-02-23_peer' \
+        --dataset tob-wgs \
+        python peer.py
+"""
+
+
+import os
+import hailtop.batch as hb
+import hail as hl
+import click
+import pandas as pd
+
+DRIVER_IMAGE = 'australia-southeast1-docker.pkg.dev/analysis-runner/images/driver:e6451763492b62ddfadc20c06b240234b20b6f2f-hail-0.2.73.devc6f6f09cec08'
+PEER_DOCKER = "australia-southeast1-docker.pkg.dev/cpg-common/images/peer:1.3"
+
+SCORES_PATH = 'gs://cpg-tob-wgs-test/kat/pca/nfe_feb22/v0/scores.json'
+COVARIATES_PATH = 'gs://cpg-tob-wgs-test/scrna-seq/grch38_association_files/covariates_files/covariates.tsv'
+SAMPLE_ID_KEYS_PATH = (
+    'gs://cpg-tob-wgs-test/scrna-seq/grch38_association_files/OneK1K_CPG_IDs.tsv'
+)
+
+
+def get_covariates(scores_path, covariates_path, sample_id_keys_path):
+    """
+    Get covariate data by merging PCA scores with age and sex info.
+    Only needs to be run once.
+    """
+    scores_df = pd.read_json(scores_path)
+    sampleid = scores_df.s
+    # only get the first 4 PCs, as indicated by scree plot
+    scores = pd.DataFrame(scores_df['scores'].to_list()).iloc[:, 0:4]
+    scores.columns = ['PC1', 'PC2', 'PC3', 'PC4']
+    scores.insert(loc=0, column='sampleid', value=sampleid)
+
+    # get age and sex from Seyhan's covariate file
+    covariates = pd.read_csv(covariates_path, sep='\t')
+    # load in keys file to convert sampleids in covariates file to CPG IDs
+    sampleid_keys = pd.read_csv(sample_id_keys_path, sep='\t')
+    covariates = pd.merge(
+        covariates, sampleid_keys, how='left', left_on='sampleid', right_on='OneK1K_ID'
+    ).drop(['pc1', 'pc2', 'pc3', 'pc4'], axis=1)
+    # merge together covariates df and scores
+    covariates = pd.merge(
+        covariates, scores, how='left', left_on='InternalID', right_on='sampleid'
+    ).drop(['OneK1K_ID', 'InternalID', 'ExternalID', 'sampleid_y'], axis=1)
+
+    return covariates.to_numpy()
+
+
+def run_peer(expression_file, covariates, factors_output_path):
```

Comment:
This is currently failing because dill is not installed. See the failed log [here](https://batch.hail.populationgenomics.org.au/batches/8273/jobs/2).

Reply:
Hoping https://github.com/populationgenomics/images/pull/4/files will unblock this PR.

Reply:
Hoping https://github.com/populationgenomics/images/pull/4 will unblock this PR.